scripts/upgrade-clients.js

```diff
@@ -0,0 +1,101 @@
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const network = hre.network.name;
+const environments = require("../environments");
+const tipMultiplier = ethers.BigNumber.from(environments.tipMultiplier);
+const tipSuggestion = "1500000000"; // ethers.js always returns this constant, it does not vary per block
+const maxPriorityFeePerGas = ethers.BigNumber.from(tipSuggestion).mul(tipMultiplier);
+const { deployProtocolClientImpls } = require("./util/deploy-protocol-client-impls.js");
+const { deploymentComplete, getFees, readContracts, writeContracts } = require("./util/utils.js");
+const Role = require("./domain/Role");
+
+/**
+ * Upgrades clients
+ *
+ * Prerequisite:
+ * - Admin must have UPGRADER role. Use `manage-roles.js` to grant it.
+ *
+ * Currently script upgrades the only existing client - BosonVoucher.
+ * If new clients are introduced, this script should be modified to get the list of clients to upgrade from the config.
+ */
+async function main(env) {
+  // Bail now if hardhat network, unless the upgrade is tested
+  if (network === "hardhat" && env !== "upgrade-test") process.exit();
+
+  const chainId = (await ethers.provider.getNetwork()).chainId;
```

Comment:
not a big deal but its easer to read

```suggestion
  const { chainId } = await ethers.provider.getNetwork();
```

Reply:
Done.